src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -560,27 +588,33 @@ def _get_enable_pod_identity(self, enable_validation: bool = False, **kwargs) ->
         """
         # read the original value passed by the command
         enable_pod_identity = self.raw_param.get("enable_pod_identity")
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if (
-            self.mc and
-            self.mc.pod_identity_profile and
-            self.mc.pod_identity_profile.enabled is not None
-        ):
-            enable_pod_identity = self.mc.pod_identity_profile.enabled
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.mc and
+                self.mc.pod_identity_profile and
+                self.mc.pod_identity_profile.enabled is not None
+            ):
+                enable_pod_identity = self.mc.pod_identity_profile.enabled
 
         # this parameter does not need dynamic completion
         # validation
         if enable_validation:
-            if enable_pod_identity and not self._get_enable_managed_identity(enable_validation=False):
-                raise RequiredArgumentMissingError(
-                    "--enable-pod-identity can only be specified when --enable-managed-identity is specified"
-                )
-            if self.mc and self.mc.network_profile and safe_lower(self.mc.network_profile.network_plugin) == "kubenet":
-                if enable_pod_identity and not self._get_enable_pod_identity_with_kubenet(enable_validation=False):
+            if self.decorator_mode == DecoratorMode.CREATE:
```

Comment:
question: what will happen if `decorator_mode` is not `CREATE`? Do we simply skip the else branch?

Reply:
Yes, we will skip all the checks in `UPDATE` mode (which is `aks update`). I suppose the validation is handled by `_update_addon_pod_identity`.